lifter/OperandUtils.cpp

```diff
@@ -92,19 +288,23 @@ Value* doPatternMatching(Instruction::BinaryOps I, Value* op0, Value* op1) {
   return nullptr;
 }
 
-KnownBits analyzeValueKnownBits(Value* value, const DataLayout& DL) {
+KnownBits analyzeValueKnownBits(Value* value, Instruction* ctxI) {
```

Comment:
cache the results? computeKnownBits is a recursive function, so its expensive. Ideally knownbit of a value is always same except if its in a basic block in a different branch.